check/shellcheck

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Summary: execute shellcheck for all shell scripts in this repository
+#
+# Usage:
+#     check/shellcheck [--dry-run] [shellcheck-arguments]
+#
+# Use the '--dry-run' option to print out the shellcheck command line without
+# running it.  This displays all shell script files found in the repository.
+
+set -uo pipefail
+
+# Change working directory to the repository root.
+thisdir=$(dirname "${BASH_SOURCE[0]:?}")
+repo_dir=$(git -C "${thisdir}" rev-parse --show-toplevel)
```

Comment:
Please either append `|| exit $?` here as in Cirq or use `set -euo pipefail` above.  With the second option you can remove `exit $?` on the line below.

Reply:
Already done in #1147  :-)

Reply:
OK, Mr. Speedy. Thanks!